### PR TITLE
[DataGrid] Fixes rows not rendering properly after height change

### LINKED ISSUE #5987: [data grid] Rows do not render after changing data and resizing with animation.
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Added Accordion to DataGrid flex container(no difference if it is passing as a Toolbar component or it is above DataGrid)  and unfolded with animation after change data (rows parameter) causes lower rows don’t render. I tried apiRef.forceUpdate() after animation, but lower rows still don’t render.

Example(https://stackblitz.com/edit/react-ts-kqudbv?file=App.tsx):

https://user-images.githubusercontent.com/53148689/187897109-fdffbed1-3e4a-4b45-8f58-a027ab1cd890.mp4

### Expected behavior 🤔

_No response_

### Steps to reproduce 🕹

Link to live example:

Steps:
1. Open: https://stackblitz.com/edit/react-ts-kqudbv?file=App.tsx.
2. Open Accordion.
3. Click “Change rows”
4. Hide Accordion
5. Lower rows don't render and appear after scrolling.

### Context 🔦

I ran into the problem when I tried to use Accordion in a custom toolbar.

### Your environment 🌎

_No response_

### Order ID 💳 (optional)

240359

### PR DESCRIPTION
**Pull request by MBilalShafi**
Fixes #5987 

Before: https://codesandbox.io/s/angry-morning-tpqfpf
After: https://codesandbox.io/s/polished-sunset-bocg83

### CODE REVIEW

File: packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx [hunk 1]
```diff
@@ -217,10 +219,12 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
 
   React.useEffect(() => {
     setContainerWidth(rootRef.current!.clientWidth);
```
Comment (flaviendelangle) [inline packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx:221]: On React 17 we did not have automatic batching, so maybe it's  better to have a shared state for the width and the height

**Comment by cherniavskii** [comment]
> Demo: https://stackblitz.com/edit/react-ts-ykvaxj?file=App.tsx

By the way, this demo still shows the reported bug, because Stackblitz ignores URLs as dependencies and uses the latest version from NPM.

To use the package built from Codesandbox CI, we have to use Codesandbox:
Before: https://codesandbox.io/s/angry-morning-tpqfpf
After: https://codesandbox.io/s/polished-sunset-bocg83

**Review by cherniavskii — APPROVED** [review]

**Comment by AndreSilva1993** [comment]
Is there any possibility of getting this one in a v5 release as well? As right now, this behaviour causes a weird user experience when resizing the datagrid. Or does it cause too much overhead since both versions' codebases look way too different from each other?

**Comment by cherniavskii** [comment]
@AndreSilva1993 I think I can port it to the `master` branch

**Comment by AndreSilva1993** [comment]
@cherniavskii that would be awesome, would appreciate you so much!

**Comment by cherniavskii** [comment]
@AndreSilva1993 Done! Will be included in the next release planned for Thursday 🙂

**Pull request merged**